splash/browser_tab.py

```diff
@@ -768,3 +793,64 @@ def log(self, message, min_level=None):
 
         message = "[%s] %s" % (self.uid, message)
         log.msg(message, system='render')
+
+
+class _LuaResume(QObject):
```

Comment:
+1 to creating such object.

I don't like that it has 'Lua' in its name, docstring, member names though: _LuaResume doesn't require Lua to work, and BrowserTab is not tied to Lua scripts in any way. If in some far future someone will add e.g. Python scripting support to Splash it should be possible to reuse this object.

This object provides a way to call one of the Python callbacks from the JS code, and ensures that

1. callback is called once, and only one callback is called;
2. allowed execution time is limited.

Any ideas for a better name? `RunOnceCallbackWrapper`?  <-- it is quite ugly, so better names are welcome :)